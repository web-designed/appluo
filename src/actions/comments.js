
import moment from 'moment'
import database from '../firebase/firebase'

//*******************************************************
// COMMENTS ACTION GENERATORS
//*******************************************************

   export const addComment = (eventId, comment) => ({
      type: 'ADD_COMMENT',
      eventId,
      comment
   })

   export const startAddComment = (eventId, {commentBody, commenter}) => {
      return(dispatch) => {
         const comment = {
            createdAt: moment().valueOf(),
            commenter,
            commentBody
         }

         return database.ref(`events/${eventId}/comments`).push(comment).then( ref => {
            dispatch(addComment(eventId, {
               ...comment,
               id: ref.key
            }))
         })
      }
   }

   export const removeComment = (eventId, commentId) => ({
      type:'REMOVE_COMMENT',
      eventId,
      commentId
   })

   export const editComment = (eventId, commentId, update) => ({
      type:'EDIT_COMMENT',
      eventId,
      commentId,
      update
   })