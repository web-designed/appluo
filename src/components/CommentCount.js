import React from 'react'

const CommentCount = ({ commentsCount }) => {
   return (
      <span>comments: <span>{ commentsCount }</span></span>
   )
}

export default CommentCount