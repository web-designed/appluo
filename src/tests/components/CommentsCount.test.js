import React from 'react'
import { shallow } from 'enzyme'
import CommentCount from '../../components/CommentCount'
import comments from '../fixtures/comments'

test('should render CommentsCount correctly', () => {
   const wrapper = shallow(<CommentCount commentsCount={comments.length} />)
   expect(wrapper).toMatchSnapshot()
})