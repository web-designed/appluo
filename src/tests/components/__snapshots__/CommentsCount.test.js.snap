// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render CommentsCount correctly 1`] = `
<span
  class="input-group-text"
>
  <img
    class="mr-1"
    src="data:image/svg+xml;base64,PD94bWwgdmVyc2lvbj0iMS4wIiA/PjxzdmcgaGVpZ2h0PSI0OCIgdmlld0JveD0iMCAwIDQ4IDQ4IiB3aWR0aD0iNDgiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyI+PHBhdGggZmlsbD0iIzZjNzU3ZCIgZD0iTTQzLjk4IDhjMC0yLjIxLTEuNzctNC0zLjk4LTRoLTMyYy0yLjIxIDAtNCAxLjc5LTQgNHYyNGMwIDIuMjEgMS43OSA0IDQgNGgyOGw4IDgtLjAyLTM2eiIvPjxwYXRoIGQ9Ik0wIDBoNDh2NDhoLTQ4eiIgZmlsbD0ibm9uZSIvPjwvc3ZnPg=="
    width="20"
  />
   
  <span
    class="badge badge-primary badge-pill"
  >
    2
  </span>
</span>
`;
